yarn-project/stdlib/src/tx/tx.ts

```diff
@@ -319,3 +330,5 @@ type HasHash = { /** The tx hash */ hash: TxHash };
 function hasHash(tx: Tx | HasHash): tx is HasHash {
   return (tx as HasHash).hash !== undefined;
 }
+
+export type TxWithHash = Tx & { txHash: TxHash };
```

Comment:
I love this type! But I'm not sure how I feel about it automagically widening the visibility of `txHash` from protected to public.. Maybe we can rename to `protected _txHash` to automatically gets assigned to `public txHash` when it's resolved?